### ISSUE TITLE
bug: LFM2 expert bias after softmax yields negative scores that affect REAP saliency ranking

### ISSUE THREAD

**Issue by egesabanci** [egesabanci/reap-mlx#93]
## Description

The LFM2 router adds `expert_bias` **after** softmax, meaning the resulting `scores` can exceed 1.0 or go negative. These biased scores are returned as `RouterResult.scores` and flow into `PruningState.accumulate`, which uses them to compute `weighted_ean_sum` and `weighted_expert_frequency_sum`.

This means the `reap` saliency formula `weighted_ean_sum / expert_frequency` incorporates the expert bias, potentially ranking an expert lower simply because its bias is negative -- even if the expert's output activations are high.

## Exact Code Citations

**src/reap/router.py, lines 219-227** (LFM2 router):
```python
gates = mx.softmax(logits, axis=-1)
if self.use_expert_bias:
    gates = gates + self.expert_bias  # <-- scores can be >1 or <0

indices = mx.argpartition(gates, kth=-self.top_k, axis=-1)[..., -self.top_k :]
scores = mx.take_along_axis(gates, indices, axis=-1)
...
return RouterResult(
    indices=indices,
    scores=scores,   # <-- biased scores passed through
)
```

**src/reap/observer.py, lines 414-420** -- these scores are accumulated:
```python
state.accumulate(
    indices=routing.indices,
    scores=routing.scores.astype(mx.float32),  # <-- biased scores
    selected_outputs=selected_outputs.astype(mx.float32),
)
```

**src/reap/metrics.py, lines 141-165** -- report() computes `reap` from `weighted_ean_sum`:
```python
"reap": (self.weighted_ean_sum / count_denominator).astype(np.float32),
```

Where `weighted_ean_sum` was accumulated using the biased scores.

## Impact

- Experts with negative expert_bias may be incorrectly pruned despite having high activation norms.
- The REAP paper's formulation weights by pure softmax probabilities, not bias-adjusted scores.
- This is documented as intentional ("matches the actual routed model computation") but should be clearer about the potential counter-intuitive effect on ranking.

## Suggested Fix

1. Document the bias impact more prominently, or
2. Provide an option to use pure softmax scores (without bias) for saliency computation while still using biased scores for actual routing.
